examples/compose/vtcompose/vtcompose_test.go

```diff
@@ -0,0 +1,454 @@
+
+	yamlString := string(finalFile)
+	assert.YAMLEq(t, referenceFile, yamlString)
+}
+
+func TestDoesNotMutateArgs(t *testing.T) {
+	baseFile := readFile(baseComposeFilePath)
+	refBaseFile := make([]byte, len(baseFile))
+	copy(refBaseFile, baseFile)
+	refKeyspaceMap := testKeyspaceInfoMap
+	refExternalDbMap := testExternalDbInfoMap
+
+	applyDockerComposePatches(baseFile, testKeyspaceInfoMap, testExternalDbInfoMap, testVtOpts)
+
+	assert.Equal(t, baseFile, refBaseFile)
+	assert.Equal(t, testKeyspaceInfoMap, refKeyspaceMap)
+	assert.Equal(t, testExternalDbInfoMap, refExternalDbMap)
+}
+
+func TestOptsAppliedThroughoutGeneratedFile(t *testing.T) {
+	baseFile := readFile(baseComposeFilePath)
+	options := vtOptions{
+		webPort:       55_555,
+		gRpcPort:      66_666,
+		mySqlPort:     77_777,
+		topologyFlags: "-custom -flags",
+		cell:          "custom cell",
+	}
+	finalFile := applyDockerComposePatches(baseFile, testKeyspaceInfoMap, testExternalDbInfoMap, options)
+	yamlString := string(finalFile)
+
+	// These asserts are not exhaustive, but should cover most cases.
+	assert.NotContains(t, yamlString, strconv.Itoa(DefaultWebPort))
+	assert.Contains(t, yamlString, strconv.Itoa(options.webPort))
+
+	assert.NotContains(t, yamlString, strconv.Itoa(DefaultGrpcPort))
+	assert.Contains(t, yamlString, strconv.Itoa(options.gRpcPort))
+
+	assert.NotContains(t, yamlString, ":"+strconv.Itoa(DefaultMysqlPort))
+	assert.Contains(t, yamlString, ":"+strconv.Itoa(options.webPort))
+
+	assert.NotContains(t, yamlString, fmt.Sprintf("-cell %s", DefaultCell))
+	assert.Contains(t, yamlString, fmt.Sprintf("-cell %s", options.cell))
+
+	assert.Contains(t, yamlString, fmt.Sprintf("- TOPOLOGY_FLAGS=%s", options.topologyFlags))
+	assert.NotContains(t, yamlString, DefaultTopologyFlags)
+}
+
+func TestDebugOptChanges(t *testing.T) {
+	baseFile := readFile(baseComposeFilePath)
+	opts := vtOptions{
+		webPort:       DefaultWebPort,
+		gRpcPort:      DefaultGrpcPort,
+		mySqlPort:     DefaultMysqlPort,
+		topologyFlags: DefaultTopologyFlags,
+		cell:          DefaultCell,
+		debug:         false,
+	}
+	debugOpts := opts
+	debugOpts.debug = true
+
+	// Debug mode.
+	finalFile := applyDockerComposePatches(baseFile, testKeyspaceInfoMap, testExternalDbInfoMap, debugOpts)
+	yamlString := string(finalFile)
+
+	assert.Contains(t, yamlString, "image: vitess/debug")
+	assert.NotContains(t, yamlString, "image: vitess/base")
+	assert.Contains(t, yamlString, "seccomp:unconfined")
+	assert.NotContains(t, yamlString, "/script/run-forever.sh $$VTROOT/bin/vtgate")
+	assert.Contains(t, yamlString, "dlv --listen=:40000 --headless=true --api-version=2 exec $$VTROOT/bin/vtgate")
+	assert.Contains(t, yamlString, `- 40000:40000`)
+	assert.NotContains(t, yamlString, "- $$VTROOT/bin/vtworker")
+	assert.Contains(t, yamlString, "- 'dlv --listen=:40001 --headless=true --api-version=2 exec $$VTROOT/bin/vtworker")
```

Comment:
What I would really like to do is assert that the a particular YAML stanza is present in the result, but I don't know of an easy way to do this and my search terms online are failing me!